_pythonSummerCourse/03-01-recursion.md

```diff
@@ -7,11 +7,181 @@ redirect_from:
 toc: false
 ---
 
-Nội dung kiến thức
+## Nội dung kiến thức
 - Các loại vòng lặp trong Python
 - Dữ liệu lồng nhau (`list` của các `list`) và vòng lặp lồng nhau
 
-Tài liệu tham khảo
+### 1. Vòng lặp `for`
+Vòng lặp `for` ở trong Python có tác dụng lặp các biến dữ liệu có trong 
+`list` , `tuple`, `string`,... Sử dụng cú pháp như sau
+```python
+for element in data:
+    # Your code
+```
+trong đó, `element` là các thành phần trong dữ liệu `data`.  
+Cụ thể, với kiểu dữ liệu `list`
+```python
+gradeList = ['K63', 'K64', 'K65', 'K66']
+for gradeName in gradeList:
+    print(gradeName)
+    
+> Output:
+K63
+K64
+K65
+K66
+```
+Ngoài ra, ta cũng có thể truy vào các thành phần của dữ liệu cần lặp thông qua 
+chỉ số của các thành phần đấy với hàm `range`
+```python
+gradeList = ['K63', 'K64', 'K65', 'K66']
+n = 4   # length of gradeList 
+for index in range(n):
+    print(gradeList[index])
+    
+> Output:
+K63
+K64
+K65
+K66
+```
+**Lưu ý:** 
+- `range(start, end, step)` là một hàm trong `Python`, đầu vào nhận các 
+**số nguyên** và trả về một chuỗi các giá trị **số nguyên** từ `start` tới `end-1`, và cách nhau `step`
+đơn vị. Ở phần code trên, ta sử dụng `range(n)` là đang mặc định `start=0, step=1`.
```

Comment:
Vì sao cần xuống dòng khi viết "đơn vị ..."?

Reply:
@QuyAnh2005 Vì sao cần xuống dòng khi viết "đơn vị ..."?

Reply:
Vì lúc viết trên Pycharm, em thấy dòng đang viết dài rồi nên em Enter xuống dòng để viết tiếp thôi ạ

Reply:
@QuyAnh2005 đọc link dưới đây và **bỏ tất cả** những dấu xuống dòng không cần thiết

https://www.jetbrains.com/help/pycharm/settings-editor-general.html#:~:text=You%20can%20also%20enable%20or,Soft%2DWrap%20Current%20Editor%20option.